dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -331,15 +332,22 @@ protected Workflow getDockstoreWorkflowById(Long id) {
         return workflow;
     }
 
+    public File downloadTargetEntry(String toolpath, ToolDescriptor.TypeEnum type, boolean unzip) throws IOException {
+        return downloadTargetEntry(toolpath, type, unzip, new File(System.getProperty("user.dir")));
+    }
+
     /**
      * Disturbingly similar to WorkflowClient#downloadTargetEntry, could use cleanup refactoring
      * @param toolpath a unique identifier for an entry, called a path for workflows and tools
      * @param unzip unzip the entry after downloading
+     * @param directory directory to unzip descriptors into
+     * @return path to the primary descriptor
      */
-    protected void downloadTargetEntry(String toolpath, boolean unzip) throws IOException {
+    public File downloadTargetEntry(String toolpath, ToolDescriptor.TypeEnum type, boolean unzip, File directory) throws IOException {
         String[] parts = toolpath.split(":");
         String path = parts[0];
-        String tag = (parts.length > 1) ? parts[1] : null;
+        // match behaviour from getDescriptorFromServer, use master if no version is provided
+        String tag = (parts.length > 1) ? parts[1] : "master";
```

Comment:
Just realized this won't work for hosted entries. Would using the most recent entry make the most sense?